gzip.go

```diff
@@ -22,8 +22,29 @@ type codings map[string]float64
 // The examples seem to indicate that it is.
 const DEFAULT_QVALUE = 1.0
 
-var gzipWriterPool = sync.Pool{
-	New: func() interface{} { return gzip.NewWriter(nil) },
+// gzipWriterPools maps compression level to a sync.Pool used to re-use instance
+// of gzip.Writer.
+var gzipWriterPools map[int]*sync.Pool
```

Comment:
Since the keys are all ints, and we know that there are always 11, it would be better to use an [11]sync.Pool here. We'll have to special-case -1, which is annoying, but I think better than doing the lookup every single time.